

arch/i386/kernel/kprobes.c: In function `setjmp_pre_handler':
arch/i386/kernel/kprobes.c:298: parse error before `unsigned'

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/i386/kernel/kprobes.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletion(-)

diff -puN arch/i386/kernel/kprobes.c~kprobes-build-fix arch/i386/kernel/kprobes.c
--- 25/arch/i386/kernel/kprobes.c~kprobes-build-fix	2004-08-25 19:18:53.240156928 -0700
+++ 25-akpm/arch/i386/kernel/kprobes.c	2004-08-25 19:18:53.243156472 -0700
@@ -292,9 +292,12 @@ int kprobe_exceptions_notify(struct noti
 int setjmp_pre_handler(struct kprobe *p, struct pt_regs *regs)
 {
 	struct jprobe *jp = container_of(p, struct jprobe, kp);
+	unsigned long addr;
+
 	jprobe_saved_regs = *regs;
 	jprobe_saved_esp = &regs->esp;
-	unsigned long addr = (unsigned long)jprobe_saved_esp;
+	addr = (unsigned long)jprobe_saved_esp;
+
 	/*
 	 * TBD: As Linus pointed out, gcc assumes that the callee
 	 * owns the argument space and could overwrite it, e.g.
_
